lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -1080,21 +1080,29 @@ bool TypeLoweringVisitor::visitDecl(MemOp op) {
             mkConnect(builder, realOldField, newField);
           } else {
             // Cast the input aggregate write data to flat type.
-            realOldField = builder->create<BitCastOp>(
-                newField.getType().cast<FIRRTLType>(), oldField);
+            auto newFieldType = newField.getType().cast<FIRRTLType>();
```

Comment:
Should we even get here?  Shouldn't 0-width components just be skipped over?

Reply:
This is the mask bit field, for the corresponding zero bit data field. So we need to ignore the corresponding mask bits which correspond to the zero bit data field.